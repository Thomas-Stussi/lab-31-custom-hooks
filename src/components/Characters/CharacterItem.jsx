import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const CharacterItem = character => (
  <>
    <Link to={`/detail/${character.id}`} >
      <figure>
        <img src={character.image} alt={character.name} />
        <figcaption>{character.name}</figcaption>
      </figure>
    </Link>
  </>
);

CharacterItem.propTypes = {
  character: PropTypes.object.isRequired
};

export default CharacterItem;
