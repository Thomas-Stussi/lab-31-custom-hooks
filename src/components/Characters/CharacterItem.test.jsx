/* eslint-disable max-len */
import React from 'react';
import { render } from '@testing-library/react';
import CharacterItem from './CharacterItem';

describe('Item Component', () => {
  it('renders Item component', () => {
    const testCharacter = {
      'id': 13,
      'name': 'Alien Googah',
      'status': 'unknown',
      'species': 'Alien',
      'type': '',
      'gender': 'unknown',
      'origin': {
        'name': 'unknown',
        'url': ''
      },
      'location': {
        'name': 'Earth (Replacement Dimension)',
        'url': 'https://rickandmortyapi.com/api/location/20'
      },
      'image': 'https://rickandmortyapi.com/api/character/avatar/13.jpeg',
      'episode': [
        'https://rickandmortyapi.com/api/episode/31'
      ],
      'url': 'https://rickandmortyapi.com/api/character/13',
      'created': '2017-11-04T20:33:30.779Z'
    };
    
    const { asFragment } = render(<CharacterItem character={testCharacter} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
