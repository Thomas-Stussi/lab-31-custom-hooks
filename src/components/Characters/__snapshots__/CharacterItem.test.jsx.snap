// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component renders Item component 1`] = `
<DocumentFragment>
  <a
    href="/detail/undefined"
  >
    <figure>
      <img />
      <figcaption />
    </figure>
  </a>
</DocumentFragment>
`;
